modules/core/src/core/attribute.js

```diff
@@ -93,50 +94,142 @@ export default class Attribute {
     }
     const buffer = this.externalBuffer || this.buffer;
     if (buffer) {
-      return [buffer, this];
+      return [buffer, this.accessor];
     }
     return null;
   }
 
-  // Sets all accessor props except type
-  // TODO - store on `this.accessor`
-  _setAccessor(opts) {
-    const {
-      // accessor props
-      size = this.size,
-      offset = this.offset || 0,
-      stride = this.stride || 0,
-      normalized = this.normalized || false,
-      integer = this.integer || false,
-      divisor = this.divisor || 0,
-      instanced,
-      isInstanced
-    } = opts;
-
-    this.size = size;
-    this.offset = offset;
-    this.stride = stride;
-    this.normalized = normalized;
-    this.integer = integer;
-
-    this.divisor = divisor;
-
-    if (isInstanced !== undefined) {
-      log.deprecated('Attribute.isInstanced');
-      this.divisor = isInstanced ? 1 : 0;
+  // PRIVATE HELPERS
+
+  _setExternalBuffer(buffer) {
+    this.externalBuffer = buffer;
+    this.constant = false;
+
+    const accessorFields = {};
+    accessorFields.type = buffer.accessor.type;
+    if (buffer.accessor.divisor !== undefined) {
+      accessorFields.divisor = buffer.accessor.divisor;
     }
-    if (instanced !== undefined) {
-      log.deprecated('Attribute.instanced');
-      this.divisor = instanced ? 1 : 0;
+
+    this.accessor = new Accessor(this.accessor, accessorFields);
+  }
+
+  _setBuffer(value, props) {
+    this.externalBuffer = null;
+    this.value = value;
+
+    if (this.gl) {
+      // Create buffer if needed
+      if (!this.buffer) {
```

Comment:
Nit: I'm surprised lint doesn't ask to merge these two ifs

Reply:
Yes, maybe it's the comment. Good since I preferred it this way...